File: LeetCode/Medium/0024-swap-nodes-in-pairs/0024-swap-nodes-in-pairs.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func swapPairs(head *ListNode) *ListNode {
    dummy := ListNode{0, nil}
        
    curr := &dummy

    for head != nil {
        if head.Next != nil {
            temp := head.Next.Next
            curr.Next = head.Next
            curr.Next.Next = head
            curr.Next.Next.Next = nil
            head = temp
        } else {
            curr.Next = head
            head = nil
        }
        curr = curr.Next.Next
    }

    return dummy.Next
}

File: LeetCode/Medium/0024-swap-nodes-in-pairs/0024-swap-nodes-in-pairs.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode swapPairs(ListNode head) {
        ListNode dummy = new ListNode();
        
        ListNode curr = dummy;
        
        while (head != null){
            if (head.next != null){
                ListNode temp = head.next.next;
                curr.next = head.next;
                curr.next.next = head;
                curr.next.next.next = null;
                head = temp;
            } else {
                curr.next = head;
                head = null;
            }
            curr = curr.next.next;
        }
        
        return dummy.next;
    }
}